packages/sdk/test/integration/quantum-policy.test.ts

```diff
@@ -0,0 +1,170 @@
+import 'reflect-metadata'
+
+import { Stream } from '../../src/Stream'
+import { StreamrClient } from '../../src/StreamrClient'
+import { StreamPermission } from '../../src/permission'
+import { FakeEnvironment } from '../test-utils/fake/FakeEnvironment'
+import { Msg } from '../test-utils/publish'
+import { createTestStream } from '../test-utils/utils'
+import { EthereumKeyPairIdentity, Identity, MLDSAKeyPairIdentity, Subscription } from '../../src'
+
+describe('Quantum encryption policies', () => {
+
+    let publisherIdentity: Identity
+    let subscriberIdentity: Identity
+    let quantumPublisher: StreamrClient
+    let quantumSubscriber: StreamrClient
+    let nonQuantumClient: StreamrClient
+    let stream: Stream
+    let environment: FakeEnvironment
+
+    beforeEach(async () => {
+        environment = new FakeEnvironment()
+
+        publisherIdentity = MLDSAKeyPairIdentity.generate()
+        subscriberIdentity = MLDSAKeyPairIdentity.generate()
+
+        // Client for setting up the stream
+        nonQuantumClient = environment.createClient({
+            auth: {
+                identity: EthereumKeyPairIdentity.generate(),
+            }
+        })
+        // nonQuantumClient also has pub/sub permissions to stream because it created the stream
+        stream = await createTestStream(nonQuantumClient, module)
+        await stream.grantPermissions({ permissions: [StreamPermission.PUBLISH], userId: await publisherIdentity.getUserIdString() })
+        await stream.grantPermissions({ permissions: [StreamPermission.SUBSCRIBE], userId: await subscriberIdentity.getUserIdString() })
+    })
+
+    afterEach(async () => {
+        await environment.destroy()
+    })
+
+    describe('pubsub under strict quantum settings', () => {
+
+        let sub: Subscription
+
+        beforeEach(async () => {
+            // Clients for pub/sub (can't make transactions)
```

Comment:
Didn't quite catch what kind of transactions are referred here? Why these client would try to do some transactions for pubsub?

Reply:
There's another client instance in this test called `nonQuantumClient` that uses an `EthereumKeyPairIdentity` which can sign transactions to create streams and set permissions. I'll clarify the comment(s).